### ISSUE TITLE
Hifigan sample config missing

### ISSUE THREAD

**Issue by DatGuy1** [CookiePPP/cookietts#13]
Not sure what you have but I'm running a test with settings here https://github.com/jik876/hifi-gan/blob/master/config_v3.json for now

**Comment by DatGuy1** [comment]
Unrelated to the issue, but I wonder if it would be good to include wavs generated from a pretrained vocoder model for the training process of the ttm models, just to see if you're still on track.

**Comment by CookiePPP** [comment]
Yep sorry, I've been waiting for audio files from a friend before uploading a proper pretrained vocoder + samples... but those samples haven't arrived in the last 7 days.

---

Feel free to use the pretrained models on jik876's repo, I don't think any of the modifications I've made will cause incompatibilities.

**Comment by DatGuy1** [comment]
Seems like the pretrained models were all on 22050Hz. Is there a way to fix the speedup-effect that results with the mismatch? (Your tacotron2_tm has 44100Hz as default sampling rate)

**Comment by CookiePPP** [comment]
```python
        filter_length  =  2048,
        hop_length     =   512,
        win_length     =  2048,
```
The other params can be scaled so 44.1Khz and 22.05Khz produce identical spectrograms. And yeah, just remembered the official HiFi-GAN uses 8000Hz as fmax.

I'll upload a checkpoint in a moment. Check back on this comment in 20~ mins.

**Comment by CookiePPP** [comment]
https://github.com/CookiePPP/cookietts/commit/b774c91a37cd57b8933d3c03c4f96c32164544f4

---

Added (untested) support for GTA.py to HiFi-GAN. 

Example command to start.
```bash
python3 train.py --checkpoint_path "cp_hifigan_universal44Khz" --config "cp_hifigan_universal44Khz/config.json" --input_training_file '/media/cookie/Samsung PM961/TwiBot/CookiePPPTTS/CookieTTS/_2_ttm/tacotron2_tm/GTAflist/map_train.txt' --input_validation_file '/media/cookie/Samsung PM961/TwiBot/CookiePPPTTS/CookieTTS/_2_ttm/tacotron2_tm/GTAflist/map_val.txt' --fine_tuning
```

I'll be testing it in another hour, I need to rerun GTA.py

**Comment by CookiePPP** [comment]
HiFi-GAN fine-tuning is tested and working on my system.
Reply if there's any problems on your end. I'll close this issue in a few days if there is no response.

Thanks for the bump!

**Comment by DatGuy1** [comment]
Just as a benchmark, how long did it take you to train a HifiGAN and Tacotron2 model? I trained LJSpeech on tacotron for 40k steps and 20k steps on Hifi GAN so far, and while the outputs are definitely decipherable they aren't really indistinguishable, perhaps due to the metallic tinge (something something fixed through a denoiser?). See [generated](https://vocaroo.com/11FrRV8Xd0ko) and [ground truth](https://vocaroo.com/1mRppMXoZdDS)

**Comment by CookiePPP** [comment]
I always warm_start from anything I can. So I don't have a number for training from scratch and I have no idea what compute you have.
On 3x 2080 Ti's it takes a couple days to train a model for audiobooks speakers, and a couple weeks for more emotive/challenging speakers.

**Comment by DatGuy1** [comment]
Training on a Tesla T4, 16gb vram, Intel Xeon Platinum 8259CL CPU, pretty standard. Keeping in mind I'm just testing out on the LJSpeech dataset, I didn't warm start tacotron since I didn't know what I could warm start it off, but I did run the command you mentioned for the vocoder. MSE loss both in training and validation is only lowering by around 0.03 per 1000 steps after around 7 hours of training. Thing is, the outputs sound fairly decent if not for the metallic sound. On WaveGlow increasing the sigma value would fix it but I'm not sure what to do for Hifi GAN.

**Comment by CookiePPP** [comment]
For HiFi-GAN you'd fine-tune it off of the postnet outputs from tacotron.

**Comment by DatGuy1** [comment]
Should I aim for a specific value loss/diagonality/etc. then? Tacotron seemed to stay at a very stable 0.42 validation loss and the alignments were fairly bright and diagonal after 1 day/13k steps of training.

**Comment by CookiePPP** [comment]
diagonality around 1.06 - 1.08. spec_MFSE and postnet_MFSE to the minimum you can during validation. Weighted score during inference to the highest value possible.

**Comment by CookiePPP** [comment]
bright alignments don't mean the model is **outputting** good spectrograms, only that it understands where in the text it is.
